src/commands/cli/versions/inspect.ts

```diff
@@ -0,0 +1,104 @@
+  `${PATH_1}/sfdx-cli-darwin-x64.tar.xz`,
+  `${PATH_1}/sfdx-cli-linux-arm.tar.gz`,
+  `${PATH_1}/sfdx-cli-linux-arm.tar.xz`,
+  `${PATH_1}/sfdx-cli-linux-x64.tar.gz`,
+  `${PATH_1}/sfdx-cli-linux-x64.tar.xz`,
+  `${PATH_1}/sfdx-cli-windows-x64.tar.gz`,
+  `${PATH_1}/sfdx-cli-windows-x64.tar.xz`,
+  `${PATH_1}/sfdx-cli-windows-x86.tar.gz`,
+  `${PATH_1}/sfdx-cli-windows-x86.tar.xz`,
+  `${PATH_2}/sfdx-linux-amd64.tar.gz`,
+  `${PATH_2}/sfdx-linux-amd64.tar.xz`,
+];
+
+interface Info {
+  origin: string;
+  version: string;
+  type: 'archive' | 'npm';
+  tag: 'latest' | 'latest-rc';
+}
+
+export default class Inspect extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+
+  public static readonly flagsConfig: FlagsConfig = {};
+
+  public async run(): Promise<Info[]> {
+    const tarDir = path.resolve(path.join('.', 'tmp', 'tar'));
+    await fs.mkdirp(tarDir);
+
+    const results: Info[] = [];
+    for (const insatllerPath of INSTALLER_PATHS) {
+      this.ux.log(`Downloading: ${cyan(insatllerPath)}`);
+      const curlResult = exec(`curl ${insatllerPath} -Os`, { cwd: tarDir });
+      if (curlResult.code !== 0) {
+        this.ux.log(red("Download failed. That's a big deal. Investigate immediately."));
+        continue;
+      }
+      const filename = path.basename(insatllerPath);
+      const unpackedDir = path.resolve(path.join('.', 'tmp', 'unpacked', filename));
+      await fs.mkdirp(unpackedDir);
+      const tarResult = exec(`tar -xf ${filename} -C ${unpackedDir} --strip-components 1`, { cwd: tarDir });
+      if (tarResult.code !== 0) {
+        this.ux.log(red('Failed to unpack. Skipping...'));
+        continue;
+      }
+      const pkgJson = (await fs.readJson(path.join(unpackedDir, 'package.json'))) as PackageJson;
+
+      results.push({
+        origin: insatllerPath,
+        version: pkgJson.version,
+        type: 'archive',
+        tag: 'latest',
+      });
+    }
+
+    const npmPkg = JSON.parse(exec('npm view sfdx-cli --json', { silent: true }).stdout) as NpmPackage;
+    results.push({
+      origin: 'npm (latest)',
+      version: npmPkg['dist-tags']['latest'] as string,
+      type: 'npm',
+      tag: 'latest',
+    });
+
+    results.push({
+      origin: 'npm (latest-rc)',
+      version: npmPkg['dist-tags']['latest-rc'] as string,
+      type: 'npm',
+      tag: 'latest-rc',
+    });
+
+    this.ux.log();
+    for (const result of results) {
+      this.ux.log(`${result.origin}: ${green(result.version)}`);
+    }
+    this.ux.log();
+    const archivesMatch = new Set(results.filter((r) => r.type === 'archive').map((r) => r.version)).size === 1;
+    this.ux.log(`All Archives Match? ${archivesMatch ? green(archivesMatch) : red(archivesMatch)}`);
+    const allMatch = new Set(results.filter((r) => r.tag === 'latest').map((r) => r.version)).size === 1;
+    this.ux.log(`All Latest Versions Match? ${allMatch ? green(allMatch) : yellow(allMatch)}`);
+    return results;
```

Comment:
does this make sense? 
```
https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz: 7.82.0
npm (latest): 7.86.3
npm (latest-rc): 7.87.0

All Archives Match? true
All Latest Versions Match? false
```
that's showing three versions. Is this just verifying that all of the archives match each other, not necessarily matching to npm's `latest`? Shouldn't there ever only be 2 versions?